wallet/src/de/schildbach/wallet/ui/send/SendCoinsViewModel.kt

```diff
@@ -124,8 +132,18 @@ class SendCoinsViewModel @Inject constructor(
             }
             .launchIn(viewModelScope)
 
-        // TODO: the coin selector will need to use CoinJoinCoinSelector if CoinJoin is ON
-        walletDataProvider.observeBalance(coinSelector = MaxOutputAmountCoinSelector())
+        coinJoinConfig.observeMode()
+            .map { mode ->
+                if (mode == CoinJoinMode.NONE) {
+                    MaxOutputAmountCoinSelector()
+                } else {
+                    MaxOutputAmountCoinJoinCoinSelector(wallet)
+                    // MaxOutputAmountCoinSelector()
+                }
+            }
```

Comment:
I'm thinking we could improve this later to avoid checking for CoinJoin mode every time we need a balance. For example, `MaxOutputAmountCoinJoinCoinSelector` can check itself if coinjoin is on, and relay the logic to `MaxOutputAmountCoinSelector` if it isn't.

Reply:
oh, that is a good idea.  I was trying to keep CoinJoin mode out of Send related View Models.